application/controllers/template_controller.py

```diff
@@ -0,0 +1,75 @@
+from jsonschema import validate, ValidationError
+from structlog import get_logger
+
+from sqlalchemy.exc import SQLAlchemyError
+from flask import current_app
+
+from application.models.models import CommunicationTemplate
+from application.utils.exceptions import InvalidTemplateException, DatabaseError
+from application.models.schema import template_schema
+
+
+logger = get_logger()
+
+UPLOAD_SUCCESSFUL = 'The upload was successful'
+PREEXISTING_TEMPLATE = 'ID already exists'
+
+
+def get_template_by_id(template_id, session):
+    try:
+        template = session.query(CommunicationTemplate).filter(CommunicationTemplate.id == template_id).first()
+    except SQLAlchemyError:
+        logger.exception("Unable to retrieve template with id: {}".format(template_id))
+        raise DatabaseError("Unable to retrieve template with id: {}".format(template_id), status_code=500)
+    return template
+
+
+def validate_template(template):
+    try:
+        validate(template, template_schema)
+    except ValidationError as exception:
+        logger.exception("Attempted to upload invalid template")
+        raise InvalidTemplateException(exception.message, status_code=400)
+
+
+class TemplateController(object):
+
+    @staticmethod
+    def upload_comms_template(template_id, template_object):
+        session = current_app.db.session()
```

Comment:
Do we need to create the session and pass it around? The docs suggest you just need to access current_app .db.session by importing it (http://flask-sqlalchemy.pocoo.org/2.3/quickstart/#a-minimal-application)